2020/3/toboggan_trajectory.rb

```diff
@@ -0,0 +1,25 @@
+TREE = '#'
+OPEN = '.'
+
+def trees_on_path(map)
+  rows_size = map.size
+  columns_size = map[0].size
+  # path is a series of slopes where each slope moves right 3 and down 1
+  # path ends when we reach last row, so there's rows_size slopes in total
+  rows = (1..rows_size - 1)
+  # columns repeat forever but we only get the initial set of columns
+  # so we have to take modulo
+  columns = (1..rows_size - 1).map { |c| c * 3 % columns_size }
+  # [[1, 3], [2, 6], ...]
```

Comment:
Cool, you make a list of coordinates you need to loop through

Reply:
yup! The `zip` method came up clutch for this (never really used it before)